core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -292,9 +294,8 @@ private[spark] class ExecutorAllocationManager(
   private def addExecutors(maxNumExecutorsNeeded: Int): Int = {
     // Do not request more executors if it would put our target over the upper bound
     if (numExecutorsTarget >= maxNumExecutors) {
-      val numExecutorsPending = numExecutorsTarget - executorIds.size
-      logDebug(s"Not adding executors because there are already ${executorIds.size} registered " +
-        s"and ${numExecutorsPending} pending executor(s) (limit $maxNumExecutors)")
+      logInfo(s"Not adding executors because our current target total " +
```

Comment:
though at some point it might be good to log it in a limited way. It will introduce another variable but at least the user will know when s/he hits the cap.